fuzz/src/chanmon_consistency.rs

```diff
@@ -704,16 +703,31 @@ pub fn do_test<Out: Output>(data: &[u8], underlying_out: Out, anchors: bool) {
 
 			let mut monitors = new_hash_map();
 			let mut old_monitors = $old_monitors.latest_monitors.lock().unwrap();
+			let mut use_old_mons = $use_old_mons;
 			for (channel_id, mut prev_state) in old_monitors.drain() {
-				monitors.insert(
-					channel_id,
-					<(BlockHash, ChannelMonitor<TestChannelSigner>)>::read(
-						&mut Cursor::new(&prev_state.persisted_monitor),
-						(&*$keys_manager, &*$keys_manager),
-					)
-					.expect("Failed to read monitor")
-					.1,
-				);
+				let serialized_mon = if use_old_mons % 3 == 0 {
+					// Reload with the oldest `ChannelMonitor` (the one that we already told
+					// `ChannelManager` we finished persisting).
+					prev_state.persisted_monitor
+				} else if use_old_mons % 3 == 1 {
+					// Reload with the second-oldest `ChannelMonitor`
+					let old_mon = prev_state.persisted_monitor;
+					prev_state.pending_monitors.drain(..).next().map(|(_, v)| v).unwrap_or(old_mon)
+				} else {
+					// Reload with the newest `ChannelMonitor`
+					let old_mon = prev_state.persisted_monitor;
+					prev_state.pending_monitors.pop().map(|(_, v)| v).unwrap_or(old_mon)
+				};
+				// Use a different value of `use_old_mons` if we have another monitor (only node B)
+				use_old_mons /= 3;
```

Comment:
One line, but I think it may need explanation for folks who aren't so much into integer math.